internal/proxyservice/tcp/pg/startup.go

```diff
@@ -26,14 +25,12 @@ func newClientOptions(options map[string]string) (co *ClientOptions, err error)
 	return
 }
 
-// Startup performs the startup handshake with the client and parses the ClientOptions.
-func (h *Handler) Startup() (err error) {
-	if h.GetConfig().Debug {
-		log.Printf("Handling connection %v", h.GetClientConnection())
-	}
+// Startup performs the startup handshake with the client and parses the clientOptions.
+func (s *SingleUseConnector) Startup() (err error) {
```

Comment:
Shouldn't be using naked returns here: https://github.com/golang/go/wiki/CodeReviewComments#naked-returns